libmamba/include/mamba/core/util_random.hpp

```diff
@@ -7,8 +7,11 @@
 #ifndef MAMBA_CORE_UTIL_RANDOM_HPP
 #define MAMBA_CORE_UTIL_RANDOM_HPP
 
+#include <algorithm>
```

Comment:
I am fine with the `cstring` include, but do we need `algorithm` and `functional`?

Reply:
Indeed, I cannot find a use for `<functional>`. Two uses of `std::generate_n` for `<algorithm>`, and I should also add `<limits>`.

Reply:
Squashing those commits.